nvi-commons/src/main/java/no/sikt/nva/nvi/common/db/NviCandidateRepository.java

```diff
@@ -105,6 +105,21 @@ public void saveNote(UUID candidateIdentifier, DbNote dbNote) {
         noteTable.putItem(newNoteDao(candidateIdentifier, dbNote));
     }
 
+    public Candidate updateCandidateAndRemoveApprovals(UUID identifier, DbCandidate dbCandidate,
```

Comment:
```

    public Candidate updateCandidateAndRemoveApprovals(UUID identifier,
                                                       DbCandidate dbCandidate,
                                                       List<DbApprovalStatus> approvals) {
        var candidate = constructCandidate(identifier, dbCandidate);
        var transaction = constructTransaction(candidate, approvals);
        client.transactWriteItems(transaction);
        return new Candidate.Builder()
                   .withIdentifier(identifier)
                   .withCandidate(dbCandidate)
                   .withApprovalStatuses(approvals) // Why are you returning these?
                   .withNotes(getNotes(identifier))
                   .build();
    }

    private TransactWriteItemsEnhancedRequest constructTransaction(CandidateDao candidate, List<DbApprovalStatus> approvals) {
        var transaction = TransactWriteItemsEnhancedRequest.builder()
                              .addPutItem(candidateTable, candidate);
        approvals.stream()
            .map(approval -> approval.toDao(candidate.identifier()))
            .forEach(approvalStatus -> transaction.addDeleteItem(approvalStatusTable, approvalStatus));
        return transaction.build();
    }

    private static CandidateDao constructCandidate(UUID identifier, DbCandidate dbCandidate) {
        return CandidateDao.builder()
                   .identifier(identifier)
                   .candidate(dbCandidate)
                   .build();
    }
    ```

Reply:
Bear in mind that this code conflicts with code elsewhere.

Line 47: constructing a CandidateDao with the constructor, not the builder. Also other places using builder.

There is another method called `constructCandidate` that does something similar